pkg/server/server.go

```diff
@@ -662,17 +671,35 @@ func createDeployment(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 		//Get from URL
-		//TODO: Duplicated code, could be moved to helper function
-		//Get JSON from url
 
 		httpClient := &http.Client{}
 
-		req, err := http.NewRequest("GET", tempJSON.PtsURL, nil)
+		ptsURL, err := url.Parse(tempJSON.PtsURL)
+		if err != nil {
+			errorMessage := fmt.Sprintf("Error parsing ptsURL\n")
+			http.Error(w, errorMessage, http.StatusInternalServerError)
+			helper.LogError.Printf(errorMessage)
+			return
+		}
+
+		internalRouterFlag := false
+
+		if ptsURL.Host == shipyardHost {
+			ptsURL.Host = internalRouterHost
+			ptsURL.Scheme = "http"
+			internalRouterFlag = true
+		}
+
+		req, err := http.NewRequest("GET", ptsURL.String(), nil)
+
+		if internalRouterFlag {
+			req.Host = shipyardHost
+			req.Header.Add("Host", shipyardHost)
+			req.Header.Add("X-ROUTING-API-KEY", base64.StdEncoding.EncodeToString([]byte(shipyardPrivateSecret)))
```

Comment:
You might need to make the `X-ROUTING-API-KEY` header name configurable via environment variable because its value is configurable in the `k8s-router`.  We can punt on this for now.